packages/server/src/utils/builders/compose.ts

```diff
@@ -188,6 +192,35 @@ const createEnvFile = (compose: ComposeNested) => {
 	writeFileSync(envFilePath, envFileContent);
 };
 
+export const processComposeFile = async (compose: ComposeNested) => {
+	const { COMPOSE_PATH } = paths();
+	const command = getProcessComposeFileCommand(compose);
+
+	if (compose.serverId) {
+		await execAsyncRemote(compose.serverId, command);
+	} else {
+		await execAsync(command, {
+			cwd: join(COMPOSE_PATH, compose.appName, "code"),
+		});
+	}
+};
+
+export const getProcessComposeFileCommand = (compose: ComposeNested) => {
+	const { composeType } = compose;
+
+	let command = "";
+
+	if (composeType === "stack") {
+		command = `export $(grep -v '^#' .env | xargs) && docker stack config -c docker-compose.yml > docker-compose.processed.yml`;
+	}
+
+	if (composeType === "docker-compose") {
+		command = "cp docker-compose.yml docker-compose.processed.yml";
+	}
```

Comment:
for this compose type I would like to remove this command, because this only creates another file for no reason

I know you want to avoid doing two validations for the file name, but I prefer to have it separate and only the stack to be processed